src/GT/DB/Usage.hs

```diff
@@ -21,7 +21,12 @@ import GT.DB.Schema.Ride
 -- TODO: esqueleto does not support CTEs. It should!
 new :: (MonadIO m) => BikeId -> RideId -> SqlPersistT m [UsageId]
 new bikeId rideId = do
-    components <- [sqlQQ|
+    mride <- get rideId
+    case mride of
+        Nothing ->
+            pure []
+        Just _ -> do
+            components <- [sqlQQ|
```

Comment:
Optimistically only doing the potentially expensive query after the ride is not existing. Probably unnecessary.